tests/unit/oidc/test_models.py

```diff
@@ -24,6 +24,15 @@ def test_check_claim_binary():
     assert wrapped("foo", "foo", pretend.stub()) is True
 
 
+def test_check_sub():
+    # Empty claims always fail.
+    assert models._check_sub(pretend.stub(), "", pretend.stub()) is False
+
+    # Malformed claims always fail.
+    assert models._check_sub(pretend.stub(), "repo", pretend.stub()) is False
+    assert models._check_sub(pretend.stub(), "repo:", pretend.stub()) is False
```

Comment:
Can you parametrize this instead?

Reply:
[3c37fb8](https://github.com/pypi/warehouse/pull/13151/commits/3c37fb85f39821c1a23f54a6557e1ddd070dd872)